ReactNativeClient/lib/components/screens/note.js

```diff
@@ -604,6 +625,101 @@ class NoteScreenComponent extends BaseScreenComponent {
 		this.setState({ noteTagDialogShown: true });
 	}
 
+	recordAudio_onPress() {
+		if (!this.state.note || !this.state.note.id) return;
+
+		if (this.state.isRecording) {
+			this.setState({ isRecording: false });
+			this.onStopRecord();
+		} else {
+			this.setState({ isRecording: true });
+			this.onStartRecord();
+		}
+	}
+
+	async onStartRecord() {
+		if (Platform.OS === 'android') {
+			try {
+				const granted = await PermissionsAndroid.request(
+					PermissionsAndroid.PERMISSIONS.WRITE_EXTERNAL_STORAGE,
+					{
+				  		title: 'Permissions for write access',
+				  		message: 'Give permission to your storage to write a file',
+				  		buttonPositive: 'ok',
+					},
+			  	);
+				if (granted === PermissionsAndroid.RESULTS.GRANTED) {
+					console.log('You can use the storage');
+				} else {
+					console.log('permission denied');
+					return;
+				}
+			} catch (err) {
+				console.warn(err);
+				return;
+			}
+		  }
+		  if (Platform.OS === 'android') {
+			try {
+				const granted = await PermissionsAndroid.request(
+					PermissionsAndroid.PERMISSIONS.RECORD_AUDIO,
+					{
+						title: 'Permissions for write access',
+						message: 'Give permission to your storage to write a file',
+						buttonPositive: 'ok',
+					},
+				);
+				if (granted === PermissionsAndroid.RESULTS.GRANTED) {
+					console.log('You can use the camera');
+				} else {
+					console.log('permission denied');
+					return;
+				}
+			} catch (err) {
+				console.warn(err);
+				return;
+			}
+		}
+		console.log('start recording');
+		const audioSet = {
+		AudioEncoderAndroid: AudioEncoderAndroidType.AAC,
+		AudioSourceAndroid: AudioSourceAndroidType.MIC,
+		AVEncoderAudioQualityKeyIOS: AVEncoderAudioQualityIOSType.high,
+		AVNumberOfChannelsKeyIOS: 2,
+		AVFormatIDKeyIOS: AVEncodingOption.aac,
+		};
+		console.log('audioSet', audioSet);
+		const uri = await this.audioRecorderPlayer.startRecorder();
+		this.audioRecorderPlayer.addRecordBackListener((e) => {
+			this.setState({
+				recordSecs: e.current_position,
+				recordTime: this.audioRecorderPlayer.mmssss(
+				Math.floor(e.current_position),
+				),
+			});
+		});
+		console.log(`uri: ${uri}`);
+	};
+
+	async onStopRecord() {
+		const result = await this.audioRecorderPlayer.stopRecorder();
+		this.audioRecorderPlayer.removeRecordBackListener();
+		this.setState({
+		  recordSecs: 0,
+		});
+		this.attachFile(
+			{
+				uri: result,
+				didCancel: false,
+				error: null,
+				type: 'audio/mp4',
+				fileName: 'joplin-'+new Date().getMonth()+'.'+new Date().getDate()+'.'+new Date().getFullYear()+' - '+new Date().getHours()+'.'+new Date().getMinutes(),
+			},
+			'audio'
+		);
+		this.setState({ isrecording: false });
+	};
```

Comment:
Could you wrap all the audio related logic in a service in lib/service?

Reply:
I don't exactly know what I'm doing, but I attempted what you have asked here.